compiler/src/iree/compiler/Codegen/Dialect/VectorExt/Transforms/VectorizeIREEVectorExtOps.cpp

```diff
@@ -549,14 +570,49 @@ vectorizeLinalgExtGatherToTransferGather(RewriterBase &rewriter,
       rewriter.getMultiDimIdentityMap(gatherTy.getRank()), padding,
       /*mask=*/Value(), inBounds);
 
+  VectorType gatherMaskType = gatherVectorTy.clone(rewriter.getI1Type());
+  Value gatherMask =
+      rewriter.create<vector::CreateMaskOp>(loc, gatherMaskType, gatherDims);
+  Value maskedGather = maskOperation(rewriter, transferGatherOp, gatherMask);
+
   // Write back into tensor.
-  auto emptyOp = rewriter.create<tensor::EmptyOp>(loc, gatherTy.getShape(),
-                                                  gatherTy.getElementType());
   SmallVector<Value> writeIndices(gatherTy.getRank(), zero);
   auto writeOp = rewriter.create<vector::TransferWriteOp>(
-      loc, transferGatherOp.getResult(), emptyOp, writeIndices);
-  rewriter.replaceOp(gatherOp, writeOp);
+      loc, maskedGather, gatherOp.getOutput(), writeIndices);
+  Value maskedWrite = maskOperation(rewriter, writeOp, gatherMask);
+
+  rewriter.replaceOp(gatherOp, maskedWrite);
   return success();
 }
 
+struct MaskedTransferGatherOpPattern : public OpRewritePattern<vector::MaskOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(vector::MaskOp maskOp,
+                                PatternRewriter &rewriter) const override {
+    auto gatherOp = dyn_cast<TransferGatherOp>(maskOp.getMaskableOp());
+    if (!gatherOp) {
+      return failure();
+    }
+    // TODO: The 'vector.mask' passthru is a vector and 'vector.transfer_read'
+    // expects a scalar. We could only lower one to the other for cases where
+    // the passthru is a broadcast of a scalar.
+    if (maskOp.hasPassthru())
+      return rewriter.notifyMatchFailure(
+          maskOp, "Can't lower passthru to vector.transfer_read");
```

Comment:
style nits

```suggestion
    if (maskOp.hasPassthru()) {
      return rewriter.notifyMatchFailure(
          maskOp, "can't lower passthru to vector.transfer_read");
    }
```

> Also, to match error message styles commonly produced by other tools, start the first sentence with a lower-case letter

https://llvm.org/docs/CodingStandards.html#error-and-warning-messages